sync/src/test_snapshot.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use hash::keccak;
+use bigint::hash::H256;
+use parking_lot::Mutex;
+use bytes::Bytes;
+use ethcore::snapshot::{SnapshotService, ManifestData, RestorationStatus};
+use ethcore::header::BlockNumber;
+
+pub struct TestSnapshotService {
+	manifest: Option<ManifestData>,
+	chunks: HashMap<H256, Bytes>,
+
+	restoration_manifest: Mutex<Option<ManifestData>>,
```

Comment:
Might be simple to have a:
```
struct Wrapped(Mutex<TestSnapshotService>);
impl SnapshotService for Wrapped {

}
```
instead of maintaining all those Mutexes/locks here.

Reply:
That would be simpler. Since this PR is already fairly large, can we save that change for another PR to revamp TestSnapshotSevice? I would like to merge this PR as quickly as possible since it is holding #6896.

There are a number of refactors that will need to happen after #6896 is merged. I would prefer to save further refactors for `TestSnapshotService` until after #6896 and #7038 are merged.